flink-cdc-composer/src/test/java/com/ververica/cdc/composer/flink/FlinkPipelineComposerITCase.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.composer.flink;
+
+import org.apache.flink.runtime.testutils.MiniClusterResourceConfiguration;
+import org.apache.flink.test.junit5.MiniClusterExtension;
+
+import com.ververica.cdc.common.configuration.Configuration;
+import com.ververica.cdc.common.pipeline.PipelineOptions;
+import com.ververica.cdc.composer.PipelineExecution;
+import com.ververica.cdc.composer.definition.PipelineDef;
+import com.ververica.cdc.composer.definition.SinkDef;
+import com.ververica.cdc.composer.definition.SourceDef;
+import com.ververica.cdc.connectors.values.ValuesDatabase;
+import com.ververica.cdc.connectors.values.factory.ValuesDataFactory;
+import com.ververica.cdc.connectors.values.sink.ValuesDataSinkOptions;
+import com.ververica.cdc.connectors.values.source.ValuesDataSourceHelper;
+import com.ververica.cdc.connectors.values.source.ValuesDataSourceOptions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.Collections;
+import java.util.List;
+
+import static com.ververica.cdc.connectors.values.source.ValuesDataSourceHelper.TABLE_1;
+import static com.ververica.cdc.connectors.values.source.ValuesDataSourceHelper.TABLE_2;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Integration test for {@link FlinkPipelineComposer}. */
+class FlinkPipelineComposerITCase {
+    private static final int MAX_PARALLELISM = 4;
+    /**
+     * Use {@link MiniClusterExtension} to reduce the overhead of restarting the MiniCluster for
+     * every test case.
+     */
+    @RegisterExtension
+    static final MiniClusterExtension MINI_CLUSTER_RESOURCE =
+            new MiniClusterExtension(
+                    new MiniClusterResourceConfiguration.Builder()
+                            .setNumberTaskManagers(1)
+                            .setNumberSlotsPerTaskManager(MAX_PARALLELISM)
```

Comment:
we need to make [ValuesDatabase class](https://github.com/ververica/flink-cdc-connectors/blob/f7dda3b67b751091f92791598ef0314bea92d09e/flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/com/ververica/cdc/connectors/values/ValuesDatabase.java#L61) load by AppClassloader to avoid unexpected results.

```
    static final org.apache.flink.configuration.Configuration configuration = new org.apache.flink.configuration.Configuration();

    static {
        configuration.set(ALWAYS_PARENT_FIRST_LOADER_PATTERNS_ADDITIONAL, Collections.singletonList("com.ververica.cdc"));
    }
```

Reply:
@lvyanquan Thanks for the solution! I think as a temporary solution this is OK. Maybe we can make some improvements for the ValuesDatabase in the future, like let it lays in an operator coordinator, and we can get talk to it via `ClusterClient#sendCoordinationRequest`